packages/transaction-controller/src/utils/simulation.ts

```diff
@@ -388,15 +403,28 @@ function getTokenBalanceTransactions(
         tokenId,
       );
 
-      result.set(simulationToken, {
+      const transaction: SimulationRequestTransaction = {
         from: request.from,
         to: event.contractAddress,
         data,
-      });
+      };
+
+      const isNFTMint =
```

Comment:
Rather than doing this check inline, should we encapsulate this more generally in an alternate private method such as `#eventHasNoPreviousBalance`?